### ISSUE TITLE
Create a snap for ubuntu

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#6822]
https://github.com/snapcrafters may be oh help

**Comment by uhoreg** [comment]
Related to https://github.com/vector-im/riot-web/issues/3718 and https://github.com/vector-im/riot-web/issues/4766

**Comment by popey** [comment]
Hey hey!  I understand riot is an electron app, right? We have [a blog post ](https://snapcraft.io/blog/bringing-electron-applications-to-linux )which details how easy it is to get a snap built. 
I'd love to help with this.

**Comment by kristbaum** [comment]
#5868 is related

**Comment by Joshndroid** [comment]
They closed the pull request March 31 citing not wanting to create another distribution. People need to make noise in issue in order for them to reconsider. Here's some noise

**Comment by aaronraimist** [comment]
Just adding a note here to say that @popey has gone ahead and created an unofficial snap 

https://twitter.com/popey/status/1133884378143166464
https://snapcraft.io/riot-web

**Comment by userj** [comment]
I tried the snap from the snapstore on Ubuntu 20.04.
It launches only on first install. After closing the app it doesn't open again. 
The signal-desktop snap had the same problem, but now that seems fixed.

**Comment by stephenjudge** [comment]
I brought up the question of the possibility of an official Snapcraft build of Element Desktop in the Matrix chatroom, before I know of this issue. There were many useful discussion points made that can be followed from this message:

https://matrix.to/#/!YTvKGNlinIzlkMTVRl:matrix.org/$1EL-t25jkkFRY_nmF3tTioeC5YWV16YD68lrizebDfo

**Comment by stephenjudge** [comment]
> I tried the snap from the snapstore on Ubuntu 20.04.
> It launches only on first install. After closing the app it doesn't open again.
> The signal-desktop snap had the same problem, but now that seems fixed.

I am using the snap build on Xubuntu 18.04.4 LTS and is working fine. Obviously I can't say about 20.04 but as it's also an LTS I can't imagine why it would stop working.

**Comment by abitrolly** [comment]
@popey the instructions from https://snapcraft.io/blog/bringing-electron-applications-to-linux produce snap with Element that freezes on run with this error.
```
FATAL:platform_shared_memory_region_posix.cc(255)] This is frequently caused by incorrect permissions on
 /dev/shm.  Try 'sudo chmod 1777 /dev/shm' to fix. 
```
I filled the issue with `electron-builder` https://github.com/electron-userland/electron-builder/issues/5574 Not sure what else can I do.

**Comment by nduja** [comment]
What do you think of taking an approach similar to the [signal-desktop](https://github.com/snapcrafters/signal-desktop/blob/master/snap/snapcraft.yaml) snap? They aren't actually building the electron app in the snap manifest, instead they're pulling down the latest .deb files and then repackaging that.

**Comment by abitrolly** [comment]
@nduja the `electon-builder` is the officially recommended way for building snaps https://snapcraft.io/docs/electron-apps so I only followed it. I will take look, thanks.

**Comment by ted-gould** [comment]
I've been using the, I guess unofficial Riot snap for a while, but it seems to be getting out-of-date. Noticeably it hasn't changed its name to Element. Though I think it is missing some video features as well.

@popey is there a PR for your snapcraft configuration that could be built on? (I didn't see one in the list) Could there be? :smile:

**Comment by popey** [comment]
@ted-gould yeah, sorry the riot-web snap got behind. It actually was built from source, not a repack of the deb like signal. Unfortunately I got somewhat frustrated and burned out by the changes which made riot-web frequently fail to build, then the name change was the final straw. I just gave up in annoyance. I haven't even looked at riot/element since the name change. If people are having trouble building using electron builder, I'd recommend bringing it up on the snapcraft forum. https://forum.snapcraft.io/

**Comment by abitrolly** [comment]
@popey already reported here https://forum.snapcraft.io/t/electron-builder-produces-freezing-snaps/22328 with no resolution so far.

**Comment by pYFZOS5** [comment]
@popey can you build a element snap for us and release to store? there're notable proprietary communication apps in store; i don't mean its bad just store needs to have lot of apps but this project seems missing then while opensource

>Unfortunately I got somewhat frustrated and burned out by the changes which made riot-web frequently fail to build, then the name change was the final straw. I just gave up in annoyance. I haven't even looked at riot/element since the name change

There're riot games you know, it's very popular then new name good. anyway i'm trying to motivate you to do that 🤩

**Comment by tgr** [comment]
There are now a few of these:
* https://snapcraft.io/element-desktop by @kenvandine (source: [kenvandine/element-desktop](https://github.com/kenvandine/element-desktop), stable: 1.10.14, edge: 1.11.1)
* https://snapcraft.io/element-client by @ArnaudLier (1.10.12, [hidden because it had issues](https://forum.snapcraft.io/t/contact-snap-owner-for-element-client-snap-uses-bad-email/31221))
* https://snapcraft.io/riot-web by @popey (1.6.0, seems also hidden)

FWIW upstream element-desktop is on 1.11.3 now; 1.11.1 is 25 days old, 1.10.14 and 1.10.12 are both from this May, 1.6.0 is from May 2020.

So https://snapcraft.io/element-desktop seems to be the way to go. (Just tried, seems to work fine.)

Some guidance about these on https://element.io/get-started#linux-details would be nice - snap apps are far superior UX for the average user comapres to messing around with apt signing keys and whatnot. If using https://snapcraft.io/element-desktop is inadvisable for some reason, the official page should probably warn about that. Otherwise it should link to it as the primary means of installation, or at least mention it.

**Comment by kenvandine** [comment]
1.11.1 of the element-desktop snap is now in stable and 1.11.3 is building to edge right now.

**Comment by kristbaum** [comment]
Could the snapcraft folder be added to the official repo, and the snap be taken over by element staff? Or maybe be moved to the snapcrafters repo?
Most users would probably prefer that a highly sensitive app like element would come from a more official source. Nothing against @kenvandine ;)

**Comment by t3chguy** [comment]
@kristbaum unlikely on the front of it being maintained by the Element Web team, we don't have sufficient bandwidth or area knowledge to take this on at this time.

**Comment by popey** [comment]
I have updated the description of riot-web to indicate it's no longer maintained and that users should instead use element-desktop.
